### PR TITLE
Rename LogVerifier::VerifyResult to LogVerifyResult.

### PR DESCRIPTION
**Pull request by pphaneuf**
There was no less than *three* things VerifyResult, one less is progress!

### CODE REVIEW

**Comment by pphaneuf** [comment]
It'd be even better if it was replaced by a `util::Status`, but one thing at a time. :wink: 

(I already had this branch lying around)

File: cpp/client/ct.cc [hunk 1]
```diff
@@ -259,7 +259,7 @@ static bool VerifySCTAndPopulateSSLClientCTData(
   sct_info->mutable_sct()->CopyFrom(sct);
   const unique_ptr<LogVerifier> verifier(GetLogVerifierFromFlags());
   string merkle_leaf;
-  LogVerifier::VerifyResult result =
+  LogVerifier::LogVerifyResult result =
```
Comment (Martin2112) [inline cpp/client/ct.cc]: Is there still a name collision on LogVerifyResult or can this be shortened?

File: cpp/log/log_verifier.h [hunk 2]
```diff
@@ -33,7 +33,7 @@ class LogVerifier {
     INVALID_MERKLE_PATH,
```
Comment (Martin2112) [inline cpp/log/log_verifier.h]: When we unify with Status we should maybe have a detail message slot in Status so we can pass the failure reason back to users.

File: cpp/client/ct.cc (hunk 1 above)
Comment (pphaneuf) [inline cpp/client/ct.cc]: It still needs to be qualified, annoyingly... But this shouldn't live too long.

File: cpp/log/log_verifier.h (hunk 2 above)
Comment (pphaneuf) [inline cpp/log/log_verifier.h]: You mean, in addition to the normal `util::Status` code and message?

Comment (Martin2112) [inline cpp/log/log_verifier.h]: Yes I meant adding a detail message to Status where appropriate, for
logging and potentially error responses to clients.

**Pull request merged**